Feature/47 did linkage contract

ITStar10 opened the pull request:
Closes #47

ITStar10 committed:
Working : Not implemented yet

ITStar10 committed:
In progress

ITStar10 committed:
Still in progress

ITStar10 committed:
Working in progress : Implementation done.

Next : Complete test code

ITStar10 committed:
Test code implemented

tahpot commented on VDA-DID-Linkage/contracts/VeridaDIDLinkage.sol:
```diff
@@ -0,0 +1,256 @@
+    /**
+     * @dev See {IVeridaDIDLinkage}
+     */
+    function addIdentifierType(string calldata identifierTypeId, string calldata signerType) external onlyOwner {
```
For efficiency should use a boolean for `signerType`. Where true = self-signed, false = trusted. Or a uint.

ITStar10 replied:
Updated

tahpot commented on VDA-DID-Linkage/contracts/VeridaDIDLinkage.sol:
```diff
@@ -0,0 +1,256 @@
+    /**
+     * @dev See {IVeridaDIDLinkage}
+     */
+    function getIdentifierList(string calldata did) external view override returns(string[] memory) {
```
Rename to `links(did)`

ITStar10 replied:
Updated

ITStar10 replied:
How about to rename to getLinks(did)?
Because, there is a function named `link()` in the contract.
There could be a bit confusion for the users if we defined 2 functions of `link()` & `links()`.

tahpot replied:
Yes, that is fine.

tahpot commented on VDA-DID-Linkage/contracts/VeridaDIDLinkage.sol:
```diff
@@ -0,0 +1,256 @@
+    /**
+     * @dev See {IVeridaDIDLinkage}
+     */
+    function getController(string calldata identifier) external view override returns(string memory) {
```
Rename to `lookup(identifier)`

ITStar10 replied:
Updated

Commit:
Add some comments and modify some error messages

tahpot commented:
@ITStar10 See my commit as I have broken your tests by changing some of the error messages. Also see other comments above.

Commit:
Update endpoints

Commit:
Add missing dependencies

ITStar10 committed:
Comments updated in the test file

Updated the string for require() statements
Test passed

Next : Updates on review

ITStar10 committed:
Updates on Review

- Updated
	SignerType updated from String to boolean
	getIdentifierList() function renamed to getLinks()
	getController() function renamed to lookup()

Test code updated & all tests passed

tahpot commented:
Thanks @ITStar10, this is good to go as well.

Can you also deploy to Mumbai and add to meta transaction server.

ITStar10 committed:
Updated deploy script
